plugins/woocommerce/includes/wccom-site/installation/installation-steps/class-wc-wccom-site-installation-step-activate-product.php

```diff
@@ -79,6 +79,11 @@ private function activate_plugin( $product_id ) {
 			throw new Installer_Error( Installer_Error_Codes::UNKNOWN_FILENAME );
 		}
 
+		// If the plugin is already active, make sure we call the registration hook.
+		if ( is_plugin_active( $filename ) ) {
```

Comment:
@vishnugopal I guess you are calling this due to `'activated_plugin'` hook not getting called from `activate_plugin` function when the plugin is already active. I went through the `activate_plugin` function to find the reason, but couldn't? Do you have any clue why?

Reply:
Ok. found it. the activation will be skipped based on `active_plugins` option.

Reply:
Yes that's right, the relevant code [is here](https://github.com/WordPress/wordpress-develop/blob/6.5/src/wp-admin/includes/plugin.php#L665). If plugin is in current list of activated plugins, the hook isn't called.

Reply:
@naman03malhotra Yes, the plugin has to be "connected" to the woocommerce.com site even if it's active. The original bug report on woocommerce.com end is here: 
19979-gh-Automattic/woocommerce.com